src/pages/HomePage/Components/ActionButtons/FilterByPriority/FilterByPriority.test.js

```diff
@@ -0,0 +1,81 @@
+import { screen, render } from "@testing-library/react";
+import "@testing-library/jest-dom";
+
+import FilterByPriority from "./FilterByPriority";
+import userEvent from "@testing-library/user-event";
+import { PRIORITY } from "../../../HomePageConstants";
+
+const mockSetFilter = jest.fn();
+
+describe("FilterByPriority", () => {
+  test("renders correctly", () => {
+    const { container } = render(
+      <FilterByPriority setFilter={mockSetFilter} filter={{}} />
+    );
+    expect(container).toMatchSnapshot();
+
+    const highButtons = screen.getAllByText("High");
+    const mediumButtons = screen.getAllByText("Medium");
+    const lowButtons = screen.getAllByText("Low");
+    const clearFilterButton = screen.getByText("Clear Filter");
+    const clearButton = screen.getByText("Clear Filter");
+
+    expect(highButtons.length).toBe(2);
+    expect(lowButtons.length).toBe(2);
+    expect(mediumButtons.length).toBe(2);
+    expect(clearFilterButton).toBeInTheDocument();
+    expect(clearButton).toBeInTheDocument();
+  });
+
+  test("activates the high filter when clicked on 'High' button", async () => {
+    render(<FilterByPriority setFilter={mockSetFilter} filter={{}} />);
+
+    const highButtonMd = screen.getByTestId("high-button-md");
+    await userEvent.click(highButtonMd);
+    expect(mockSetFilter).toHaveBeenCalledWith(PRIORITY.HIGH);
+
+    const highButtonSm = screen.getByTestId("high-button-sm");
+    await userEvent.click(highButtonSm);
+    expect(mockSetFilter).toHaveBeenCalledWith(PRIORITY.HIGH);
+  });
+
+  test("activates the medium filter when clicked on 'Medium' button", async () => {
+    render(<FilterByPriority setFilter={mockSetFilter} filter={{}} />);
+
+    const mediumButtonMd = screen.getByTestId("medium-button-md");
+    await userEvent.click(mediumButtonMd);
+    expect(mockSetFilter).toHaveBeenCalledWith(PRIORITY.MEDIUM);
+
+    const mediumButtonSm = screen.getByTestId("medium-button-sm");
+    await userEvent.click(mediumButtonSm);
+    expect(mockSetFilter).toHaveBeenCalledWith(PRIORITY.MEDIUM);
```

Comment:
Add snapshot matching in other test cases as well if you forsee the snapshot being different